Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiCpuProfiler.inl

```diff
@@ -51,110 +51,165 @@ namespace AZ
 
             const ImVec2 windowSize(900.0f, 600.0f);
             ImGui::SetNextWindowSize(windowSize, ImGuiCond_Once);
-            bool captureToFile = false;
-            if (ImGui::Begin("Cpu Profiler", &keepDrawing, ImGuiWindowFlags_None))
+            if (ImGui::Begin("CPU Profiler", &keepDrawing, ImGuiWindowFlags_None))
             {
-                m_paused = !AZ::RHI::CpuProfiler::Get()->IsProfilerEnabled();
-                if (ImGui::Button(m_paused ? "Resume" : "Pause"))
-                {
-                    m_paused = !m_paused;
-                    AZ::RHI::CpuProfiler::Get()->SetProfilerEnabled(!m_paused);
-                }
-
-                // Update region map and cache the input cpu timing statistics when the profiling is not paused
+                // Collect the last frame's profiling data
                 if (!m_paused)
                 {
-                    UpdateGroupRegionMap();
+                    // Update region map and cache the input cpu timing statistics when the profiling is not paused
                     m_cpuTimingStatisticsWhenPause = currentCpuTimingStatistics;
+
+                    CollectFrameData();
+                    CullFrameData(currentCpuTimingStatistics); 
+
+                    // Only listen to system ticks when the profiler is active
+                    if (!SystemTickBus::Handler::BusIsConnected())
+                    {
+                        SystemTickBus::Handler::BusConnect();
+                    }
                 }
 
-                if (ImGui::Button("Capture"))
+                if (m_enableVisualizer)
                 {
-                    captureToFile = true;
+                    DrawVisualizer();
                 }
-
-                if (!m_lastCapturedFilePath.empty())
+                else
                 {
-                    ImGui::SameLine();
-                    ImGui::Text(m_lastCapturedFilePath.c_str());
+                    DrawStatisticsView();
                 }
+            }
+            ImGui::End();
+         
+            if (m_captureToFile)
+            {
+                AZStd::sys_time_t timeNow = AZStd::GetTimeNowSecond();
+                AZStd::string timeString;
+                AZStd::to_string(timeString, timeNow);
+                u64 currentTick = AZ::RPI::RPISystemInterface::Get()->GetCurrentTick();
+                AZStd::string frameDataFilePath = AZStd::string::format("@user@/CpuProfiler/%s_%llu.json", timeString.c_str(), currentTick);
```

Comment:
This might be better as
```
const AZStd::string frameDataFilePath = AZStd::string::format(
    "@user@/CpuProfiler/%s_%llu.json",
    timeString.c_str(),
    AZ::RPI::RPISystemInterface::Get()->GetCurrentTick());
```